metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/SearchTerm.java

```diff
@@ -1,14 +1,35 @@
 package eu.europeana.enrichment.api.internal;
 
 import eu.europeana.enrichment.utils.EntityType;
-import java.util.List;
+import java.util.Set;
 
+/**
+ * Represents a search term: a text reference (with an optional language) in a record with a list of
+ * candidate reference types. Two search terms are equal if and only if they have the same reference
+ * (defined by text and language) and candidate types.
+ */
 public interface SearchTerm {
 
-  List<EntityType> getCandidateTypes();
+  Set<EntityType> getCandidateTypes();
 
   String getTextValue();
 
   String getLanguage();
 
+  /**
```

Comment:
Equals and hashCode shouldn't be in the interface imo.
The implementer should decide if the equals should be applied on the implementing class or not.
Btw they were removed on a previous review.